.gitmodules

```diff
@@ -268,3 +268,6 @@
 [submodule "contrib/hashidsxx"]
 	path = contrib/hashidsxx
 	url = https://github.com/schoentoon/hashidsxx.git
+[submodule "contrib/base-x"]
+	path = contrib/base-x
+	url = https://github.com/Kronuz/base-x.git
```

Comment:
To be totally honest, I am a little bit skeptical about the base-x library. It has one contributor and the last commit was five years ago. This means it is basically dead and if there shall ever be a problem, we are on our own.

If there is another base en/decoding C++ library which can provide the needed functionality, it would be great if we could use that instead.

Reply:
Well, at first that confused me, too. I looked through many of the present modules on GH, and they all do the same. 
This module actually works (what else to commit there?), and it also can work with other encodings -- we may use it afterwards. The only thing I'm concerned about -- maybe this lib shall be forked just in case it is deleted one time.

Reply:
Chances are that the original author already forget about his creation and will not delete it 😄 But yes, to be on the safe side, a fork in the ClickHouse GitHub organization makes sense. Here you go: https://github.com/ClickHouse/base-x

Reply:
"Base-x" on GitHub an umbrella term for the same functionality in different languages ... I found Java, PHP, JavaScript etc. implementations.

But there really seems no well-maintained and C++ base58 library on GitHub. Just found https://github.com/bitcoin/libbase58 which is C and doesn't look very active. So, in that case we may go with base-x.

Reply:
Yes
And moreover and very important
All other libs have only ONE alphabet hardcoded, but we need the 4 most popular